lib/mongoid/clients.rb

```diff
@@ -117,8 +117,11 @@ module ClassMethods
       def mongo_client
         name = client_name
         client = Clients.with_name(name)
-        client = client.use(database_name)
-        client.with(self.persistence_options)
+        if self.persistence_options
+          client.with(self.persistence_options.merge(database: database_name))
+        else
+          client.use(database_name)
```

Comment:
Mongo::Client#use actually just calls Mongo::Client#with
https://github.com/mongodb/mongo-ruby-driver/blob/master/lib/mongo/client.rb#L207

so do you want to change this to the following? It might end up being a tiny bit faster.
```ruby
client.with(database: database_name)
```